sktime/markov/hmm/_bindings/include/utils.h

```diff
@@ -228,21 +211,60 @@ void backward(const np_array<dtype> &transitionMatrix, const np_array<dtype> &po
             for (i = 0; i < N; ++i) {
                 sum = 0.0;
                 for (j = 0; j < N; ++j) {
-                    sum += transitionMatrixPtr[i * N + j] * pObsPtr[t * N + j] * betaPtr[t * N + j];
+                    sum += transitionMatrix[i * N + j] * pobs[t * N + j] * beta[t * N + j];
                 }
-                betaPtr[(t - 1) * N + i] = sum;
+                beta[(t - 1) * N + i] = sum;
                 scaling += sum;
             }
             // scale this row
             if (scaling != 0) {
                 for (j = 0; j < N; ++j) {
-                    betaPtr[(t - 1) * N + j] /= scaling;
+                    beta[(t - 1) * N + j] /= scaling;
                 }
             }
         }
     }
 }
 
+template<typename dtype>
+void backward(const np_array<dtype> &transitionMatrix, const np_array<dtype> &pobs, np_array<dtype> &beta,
+              const py::object &pyT) {
+    std::size_t T = [&pyT, &pobs]() {
+        if (pyT.is_none()) {
+            return static_cast<std::size_t>(pobs.shape(0));
+        } else {
+            return py::cast<std::size_t>(pyT);
+        }
+    }();
+
+    if (beta.ndim() != pobs.ndim() || static_cast<std::size_t>(beta.shape(0)) < T || beta.shape(1) != pobs.shape(1)) {
+        throw std::invalid_argument("Shape mismatch: Beta must have at least size T and otherwise shape of beta must "
+                                    "match shape of state probability trajectory.");
```

Comment:
i don't totally understand this error message. "beta must have at least size T" i understand as, if beta's shape is T's shape or bigger, then it's fine
"otherwise shape of beta must match shape of state probability trajectory" - this i don't understand. "otherwise" as in, if beta's shape is smaller than T's shape? or if it is T's shape or bigger this is an additioanl requirement?

Reply:
what it should say is: the number of dimensions should match, the length of beta should be at least T (ie `beta.shape[0] >= T`) and in the remaining dimensions it should match the shape of the state probability trajectory, i.e. `beta.shape[i] == pobs[i]` for `i > 0`.

Reply:
this is much more clear to me :) hit me up on slack if you want to go back and forth on a revised error message